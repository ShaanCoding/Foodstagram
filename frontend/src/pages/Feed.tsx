import UseFeedQuery from '../api/UseFeedQuery'
import useAuth from '../api/util/useAuth'
import Spinner from '../components/common/Spinner'
import { Post } from '../components/post/Post'

const Feed = () => {
	const [account, isLoading] = useAuth()
	const feedQuery = UseFeedQuery()

	return (
		<div>
			{feedQuery.isLoading ||
				(isLoading && (
					<div className="h-[600px]">
						<Spinner />
					</div>
				))}
			{feedQuery.isLoading === false &&
				feedQuery.isSuccess &&
				feedQuery.data.data.posts === undefined && (
					<h1 className="mx-auto block w-2/3 text-center mt-[200px] mb-[200px]">
						Your feed appears to be empty, we recommend you follow some more
						people!
					</h1>
				)}

			{!feedQuery.isLoading &&
				feedQuery.isSuccess &&
				feedQuery.data.data.posts !== undefined &&
				feedQuery.data?.data.posts.map((post, index) => (
					<Post key={index} post={post} />
				))}
		</div>
	)
}

export default Feed
