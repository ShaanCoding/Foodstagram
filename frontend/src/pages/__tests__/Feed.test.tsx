import { QueryClient, QueryClientProvider } from 'react-query'
import { BrowserRouter, Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Feed from '../Feed'

const queryClient = new QueryClient()

it('The Feed page renders correctly', () => {
	const tree = renderer
		.create(
			<QueryClientProvider client={queryClient}>
				<BrowserRouter>
					<Feed />
				</BrowserRouter>
			</QueryClientProvider>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
