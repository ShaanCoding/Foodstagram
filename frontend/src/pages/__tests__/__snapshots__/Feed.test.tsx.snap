// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Feed page renders correctly 1`] = `
<div>
  <p
    className="text-sm text-black-500 text-left inline-block align-middle mb-4"
  >
    No posts yet! Try following some users.
  </p>
</div>
`;
